torch/utils/data/dataloader.py

```diff
@@ -293,7 +303,7 @@ def _shutdown_workers(self):
             self.shutdown = True
             self.done_event.set()
             # if worker_manager_thread is waiting to put
-            while not self.data_queue.empty():
+            while self.data_queue is not None and not self.data_queue.empty():
```

Comment:
When can this queue be `None`?

Reply:
When there is an error from `worker_init_fn` as it can happen right after `w.start()` but before setting `self.data_queue`.

Reply:
Sorry I don't see what you mean. `worker_init_fn` is executed in the worker, while this code runs in the main process.

Reply:
It is related to the SIGCHLD handler thing. When a worker exits from `worker_init_fn` before we set `data_queue` after line 215 above, this method will be called and an `AttributeError` will be thrown. So I added this and a dummy `self.data_queue=None` before setting workers.

However, I'm moving the `for w in self.workers: w.start()` to after setting `self.data_queue`. So this check will be removed in my new commit, if I don't find any more issues. :)